File: output/gpt_4o/avatar/algo_based_translation/python/java/codeforces_624_B.java

import java.util.*;
import java.io.*;

public class codeforces_624_B {
    public static void main(String[] args) throws IOException {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(reader.readLine().trim());
        int[] a = Arrays.stream(reader.readLine().trim().split(" "))
                        .mapToInt(Integer::parseInt)
                        .toArray();
        
        double pre = Double.POSITIVE_INFINITY;
        int ans = 0;

        Arrays.sort(a);
        for (int j = n - 1; j >= 0; j--) {
            int maxValue = (int) Math.max(0, Math.min(pre - 1, a[j]));
            ans += maxValue;
            pre = maxValue;
        }
        
        System.out.println(ans);
    }
}

File: output/deepseek_r1/avatar/algo_based_translation/java/python/successful_translation/codeforces_624_B.py

import sys

def main():
    n = int(sys.stdin.readline())
    ans = list(map(int, sys.stdin.readline().split()))[:n]
    ans.sort()
    count = ans[-1]
    for i in range(len(ans)-2, -1, -1):
        if ans[i] < ans[i+1]:
            count += ans[i]
        else:
            while ans[i] >= ans[i+1]:
                ans[i] -= 1
            if ans[i] > 0:
                count += ans[i]
    print(count)

if __name__ == "__main__":
    main()

